File: src/event/impl/player/PlayerDisconnectEvent.php
<?php

namespace pocketcloud\cloud\event\impl\player;

use pocketcloud\cloud\player\CloudPlayer;
use pocketcloud\cloud\server\CloudServer;

class PlayerDisconnectEvent extends PlayerEvent {

    public function __construct(
        CloudPlayer $player,
        protected readonly ?CloudServer $server,
        protected readonly string $serverName
    ) {
        parent::__construct($player);
    }

    public function getServer(): CloudServer {
        return $this->server;
    }

    public function getServerName(): string {
        return $this->serverName;
    }
}

File: src/main/java/de/pocketcloud/cloud/event/impl/player/PlayerDisconnectEvent.java
package de.pocketcloud.cloud.event.impl.player;

import de.pocketcloud.cloud.player.CloudPlayer;
import de.pocketcloud.cloud.server.CloudServer;
import lombok.Getter;

public final class PlayerDisconnectEvent extends PlayerEvent {

    @Getter
    private final CloudServer server;
    @Getter
    private final String serverName;

    public PlayerDisconnectEvent(CloudPlayer player, CloudServer server, String serverName) {
        super(player);
        this.server = server;
        this.serverName = serverName;
    }
}
